pathlength functions

kesshijordan opened the pull request:
making path length functions
making an affine mapping from voxel indices to tvis points, regardless of voxel order

RafaelNH committed:
NF: Adding file for reconstruction of the free water diffusion tensor model

RafaelNH committed:
NF, DOC: Code for the FreeWaterTensorModel class was edited. Documentation of the class was updated

RafaelNH committed:
NF: Start editing the WLLS fit of the Free Water Elimination Model basing on the most recent version of the WLLS DTI fit of the DTI module dipy.reconst.dti

RafaelNH committed:
BF, DOC and start adding some optional parameters for the WLS fit

RafaelNH committed:
Adding some code to the WLS water elimination fit (WIP)

RafaelNH committed:
Adding more code to the WLS water elimination fit - function should not be working yet - each line will be corrected using tests

RafaelNH committed:
RF: add helpers function _wls_iter which process the water elimination model at the voxel level

RafaelNH committed:
TEST, BF: Using the test in dipy/reconst/tests/test_fwdti.py bugs on fwdti.py were fixed. Now we have a functional version of the free water elimination model, however its inputs are not that I would expect.

RafaelNH committed:
BF: Problem with the code was found - WLS fit is now working as expected. Test passes with the expected precision

RafaelNH committed:
NF: Code for the prediction function @etpeterson input. A test still have to be added to test the prediction

RafaelNH committed:
NF: Add the water volume fraction f to the FreeWaterTensorFit class properties

RafaelNH committed:
TEST, BF: A test for the volume fraction estimate was added. Bug in FreeWaterTensorModel was found and corrected.

RafaelNH committed:
TEST, BF: Add a test to evaluate different algorithm precision. Bug found and corrected

RafaelNH committed:
Test: Testing multi-voxel data and random volume fractions

RafaelNH committed:
RF: adding a min signal correction to the water free adjusted signal

RafaelNH committed:
TEST: WLS procedure can be instable for some f, therefore I decresed the precision of array comparison of test test_fwdti_multi_voxel

RafaelNH committed:
TEST, BF: First version of the fwdti_prediction test (still single voxel and with low precision). Bug in function fwdti_prediction found and fix

RafaelNH committed:
RF, DOC: log(1) is 0 and correction of a typo in documentation (corrections suggested by @samuelstjean)

RafaelNH committed:
RF TEST: WLS model is failing when f ground truth is larger or equal to 0.9. For now I am romoving this cases in test since this can be a situation that the algorithm intrinsically cannot solve (see article). However I will improve the procedure by incorporating f=0 and f=1 in tests and see if the algorithm already can handle this cases

RafaelNH committed:
TEST: Test prediction functions in model and fit classes

RafaelNH committed:
TEST: Insure that simulations and espected values are always estimated relative to the same axis (for this I selected the x-axis)

RafaelNH committed:
TEST, BF, RF: fw_prediction is now able to predict the signal for multi-voxel parameters.

RafaelNH committed:
NF: Adding the non-linear water elimination fit (WIP)

RafaelNH committed:
RF: Adding S0 as a model parameter (I will need this for the non-linear fitting)

RafaelNH committed:
NF: First version of the non-linear fit implemented. Using this fit we can conver the full range of possible f values

RafaelNH committed:
Test, BF: Add a test module to evaluate the expected perfomance if a wrong mask, method fit, and min_signal are introduced to the FreeWaterTensorModel. According to this module a bug was found and fixed. I also ajusted the NLS as the default method

RafaelNH committed:
TESTS: Test the restore procedure using grounth truth free noise signals. Testing errors if sigma is not given but user is calling NLS sigma weigthing.

RafaelNH committed:
DOC, NF: Add a first version of the example usage script. At the moment this script is timing the fwdti model performance when applied to a slice of the CENIR MULTIB dataset. Also added S0 as an property of FreeWaterTensorFit class.

RafaelNH committed:
DOC, PEP8

RafaelNH committed:
TEST, BF: NLS procedure with given initial guess was not working properly for multi voxel data (initial guess matrix has to be flatten as the data)

RafaelNH committed:
NF: Add two functions to convert tensor elements to cholesky elements and vice versa. Tested with a positive defined matrix

RafaelNH committed:
NF: Add two functions to convert tensor elements to cholesky elements and vice versa. Tested with a positive defined matrix

RafaelNH committed:
NF: Add cholesky decomposition in non-linear fwDTI fit procedure to insure that the diffusion tensor is positive define. Apply transformation to water volume fraction on the same procedure to insure volume fraction is estimated in the range between 0 and 1 (sine transformation was used for this ftrans = asin(2*f - 1) + pi/2 <=> f = 0.5*sin(ftrans - pi/2) + 0.5 so that for ftrans in ]-inf, inf[ f is always between 0 and 1

RafaelNH committed:
NF: Add cholesky decomposition in non-linear fwDTI fit procedure to insure that the diffusion tensor is positive define. Apply transformation to water volume fraction on the same procedure to insure volume fraction is estimated in the range between 0 and 1 (sine transformation was used for this ftrans = asin(2*f - 1) + pi/2 <=> f = 0.5*sin(ftrans - pi/2) + 0.5 so that for ftrans in ]-inf, inf[ f is always between 0 and 1

RafaelNH committed:
RF: wlls_fit_tensor comes after _wls_iter

RafaelNH committed:
RF: Adding f transformation as a separate option

RafaelNH committed:
RF: Making the code easier to read

RafaelNH committed:
NF, RF: wlls has now an optional parameter S0. If user gives a value of S0 this parameter is not taken as a free models parameter - wls performance seems to be more robust if this is given. weights are now defined from signals - in this case riterations did not show to improve the performance of the procedure so its option was removed

RafaelNH committed:
BW: Comment out scipy.optimize.least_square since it did not overcome performance of scipy.optimize.leastsq when f_transform=True and cholesky=True. Comment lines will be removed after feedback of other developers

RafaelNH committed:
NF, Test: Add the input step in the prediction function of fwdti fit class. This will prevent memory issues (as suggested by @arokem I used a strategy similar to #857). Prediction tests were added to access the performance of the new version of the class fwdti fit prediction function

RafaelNH committed:
PEP8: fwdti.py and test_fwdti.py

RafaelNH committed:
RF: @etpeterson initial guess check suggestion for the NLS solver

RafaelNH committed:
DOC: correcting typos

RafaelNH committed:
NF: Add jacobian in the non-linear procedure

souravsingh committed:
Update to the import

souravsingh committed:
Update the import statement

souravsingh committed:
Updated the import statement

souravsingh committed:
Updated the import statement

souravsingh committed:
Update the import statement

RafaelNH committed:
RF: Remove the commeted lines of code regarding the use of least_squares scipy's function since it uses didn't should to be advantagous to this problem - I will create a repository with more details on this - a promise

RafaelNH committed:
NF: Add dti md regularization of the WLS procedure's fw_params. If md is
almost near the free water diffusion value, it will be assumed that the
voxel only contains free water diffusion (i.e. f=1). Other parameters are
set to zero. This is to avoid that the tissue component of the fwdti model
overfits a true free water diffusion voxel. If this was not done, tissue
component overfitting could produce wrong low f values for voxels
containing only free water diffusion.

RafaelNH committed:
RF: refactoring (loads of refactoring) so that S0 is not taken as an index in model_params. Now S0 is an separate output variable (this was based on the suggestion done by @arokem). S0 is now also an optional parameter in prediction and fit functions. If not given in the prediction functions it is assumed to be 1, while if not given in fit functions it is estimated from dti models. I also added the MD regularization variable in the non-linear procedure. Last, I changed the initial guess of the non-linear procedure to standard wls dti since I observed that the recent version of the non-linear procedure is relatively robust to the initial guess (this is not the case when the jacobian is set to True - in this last case users should pre-calculate a better initial guess - using for example the WLS procedure)

RafaelNH committed:
NF: Changed my mind - since defining S0 from standard DTI cannot be adequate from jac=True, I decided to replace the default initial guess to the solution given by fw DTI WLS procedure.

RafaelNH committed:
RF: Remove the unused variable inv_desing in WLS procedure

RafaelNH committed:
RF: Remove the unused variable inv_desing in WLS procedure

RafaelNH committed:
BF: fwDTI fit functions make now a copy of parameters initial guess instead of changing them with the final estimates

RafaelNH committed:
DOC, RF: Improve functions documentation and add min_signal as a variable in _wls_iter

RafaelNH committed:
RF: Remove absolete imports

RafaelNH committed:
NF, TEST: Function raise error if only 1 non-zero bvalue is used

RafaelNH committed:
STYLE: Rename nlls methods as nls (more consistent to wls)

RafaelNH committed:
STYLE: use absolute imports

RafaelNH committed:
BF, DOC: Insure model parameters output has 13 elements and data is flat during wls and nls processing

RafaelNH committed:
BF, STYLE: Bug in NLS parameters reshape was fixed. Rename variables to make code easier to read

arokem committed:
RF: Speed up ODF calculation (by not calculating it in empty regions)

arokem committed:
White space.

arokem committed:
BF: masking is not needed here.

arokem committed:
Change dti profiling script to run on larger amounts of data

arokem committed:
A few more tweaks.

RafaelNH committed:
PEP8, DOC: PEP8 and correction of typos

According to the comments made by @Garyfallidis and @arokem

RafaelNH committed:
DOC: docstring and minor changes in documentation

matthieudumont committed:
Base workflow enhancements + tests

matthieudumont committed:
Fixed some docstrings, added shortnames (for combined workflows params

matthieudumont committed:
Removed unused test_utils

matthieudumont committed:
Disabled this wya of testing scripts

matthieudumont committed:
Fixed some python3 errors

matthieudumont committed:
More python3 fixes

matthieudumont committed:
Newer version of get_io_iterator compatible with all pythons.

matthieudumont committed:
Reabased multiple review fixes

matthieudumont committed:
Added dipy.io.image for easy image loading

matthieudumont committed:
Now using dipy.image.io save/load images

matthieudumont committed:
Review comments - added returns in docstrings

matthieudumont committed:
Bunch of docstring fixes

matthieudumont committed:
Changed last_generated_outputs management to a more user friendly version

matthieudumont committed:
Fix force param

matthieudumont committed:
Added the executable for denoising

matthieudumont committed:
Fix failing test

MrBago commented on dipy/tracking/utils.py:
```diff
@@ -956,3 +958,123 @@ def reduce_rois(rois, include):
+        if len(each) > 1:
+            yield each
+
+def as_segments(streamline, break_points):
```
Should we make these private functions, ie `_part_segments`?

MrBago commented on dipy/tracking/utils.py:
```diff
@@ -956,3 +958,123 @@ def reduce_rois(rois, include):
+
+    sl_vox_order = tvis_hdr['voxel_order']
+    voxel_size = tvis_hdr['voxel_size']
+    dim = nii_data.shape
```
Do we need `nii_data` as an input? Can we get this info from `tvis_hdr['dim']`?

RafaelNH committed:
PEP8

RafaelNH committed:
DOC: Editting the example intro

RafaelNH committed:
DOC: Add free water DTI formula in example

RafaelNH committed:
DOC: Edit example's data load and display

matthieudumont committed:
Removed deprecated use of getargspec in python > 3

quantshah committed:
Added fill_na function

quantshah committed:
Test for leastsq failing

RafaelNH committed:
DOC: Finish a first version of the free water DTI usage example

matthieudumont committed:
Fixed shebangs, added nlmeans to setup

quantshah committed:
Changed fill_na. Tests

quantshah committed:
Only fill for specific bounds violation

RafaelNH committed:
DOC: Changes on the example of usage
acording to the comments made by @arokem

RafaelNH committed:
RF: Remove duplicated code of the two conditions of _wls_iter

nilgoyette committed:
Merge branch 'master' of https://github.com/nipy/dipy into target_line_based

nilgoyette committed:
Fix typos, better doc, minor fixes.

RafaelNH committed:
RF: Remove step input of function predict of the FreeWaterTensorFit

RafaelNH committed:
RF: Error when data with one non zero b-value is given
now this error will raise at the inicialization of the FreeWaterTensorModel. Before this error was only raised when data was fitted

RafaelNH committed:
RF: Using the multi_voxel_fit decorator
for this some additional changes were required: 1. S0 estimations is not
required as input, instead it gets the average of b-value=0 signals. 2) S0
is not given as output. 3) NLS always uses WLS as initial guess (the
option of giving a different initial guess had to be removed). 4)
min_signal cannot be defined as min signal of the all dataset, now it is
defined as an optional variable with default value of 1e-6

quantshah committed:
Removed fill_na. Added bounds checking in fitting with warning

quantshah committed:
Added indexing test and warning for Scipy version > 0.17

quantshah committed:
Added warning message for leastsq failing

RafaelNH committed:
BF: Fix tests and add code to deal with background voxels

arokem committed:
TST: Test for invariance of model_params to splitting of the data.

This looks like a bug to me: it shouldn't matter whether you fit the
entire data array or just one.

This happens because min_signal currently depends on the entire data array
that is passed, and that varies here depending on whether only one voxel
or the entire array is passed.

arokem committed:
BF: Set the minimum positive signal to be a constant.

arokem committed:
BF: Update DKI to use the new dti module constant.

RafaelNH committed:
RF: Update the functions nls_fit_tensor and wls_fit_tensor
Now these functions are standalone function and still need to be tested

Garyfallidis committed:
Merge pull request #1112 from nipy/sfm-doc-math

DOC: Math rendering issue in SFM gallery example.

Garyfallidis committed:
Merge pull request #1087 from matthieudumont/enhanced_workflows_base

Base workflow enhancements + tests

RafaelNH committed:
DOC: Add description of results and add problematic voxels correction in example of usage

RafaelNH committed:
Test: Now FA and MD values are tested
before test were almost based only on f values

RafaelNH committed:
BF: Correct min_signal bug

RafaelNH committed:
RF: simplify min_signal and avoid flatting data

RafaelNH committed:
TEST: Add tests for the standalone functions

RafaelNH committed:
TEST, BF: Add a test for the md regularization
based on this test a bug was fixed (params were not in the format assumed by function mean_diffusivity)

RafaelNH committed:
DOC: correct some typos and order function inputs arguments

Garyfallidis committed:
Merge pull request #1049 from grlee77/vfields

MAINT: minor cython cleanup in align/vector_fields.pyx

Garyfallidis committed:
Merge pull request #1046 from gabknight/BF_nbr_process_le_zero

BF - peaks_from_model with nbr_processes <= 0

RafaelNH committed:
DOC: Add return section in predict functions

RafaelNH committed:
DOC minor change in documentation multi -> multiple b-values

Garyfallidis committed:
Merge pull request #835 from RafaelNH/free_water_DTI

NF: Free water tensor model

Garyfallidis committed:
Merge pull request #1075 from arokem/drop26

Drop26

quantshah committed:
Test version with LooseVersion

quantshah committed:
Removed tests for warning

Commit:
adding tutorial1.py

quantshah committed:
Added conditional test

Commit:
add test

kesshijordan committed:
fixed colon spacing in docstring

MarcCote committed:
BF: should stop the occasional segmentation fault we had when closing the vtkWindow

MarcCote committed:
Renamed record and play methods in ShowManager

arokem committed:
BF: Time elapsed needs to be at least 1 second for there to be a difference.

arokem committed:
PEP8 + DOC: A few fixes in the workflows module.

arokem committed:
Remove un-needed function.

arokem committed:
Restore docstring to not have this question.

arokem committed:
Add clarifying comment.

arokem committed:
TST: Remove test of un-necessary function.

arokem committed:
Whitespace (PEP8).

kesshijordan committed:
Merge remote-tracking branch 'upstream'

matthieudumont committed:
Add the reconst_dti workflow

Commit:
add intensity adjustment

Commit:
tutorial

RafaelNH committed:
BF: Deal with negative S0
in teory we shouldn't have negative S0, however this can happen in partice. For example this case was found in the CENIR HCP like dataset

matthieudumont committed:
Fixed formatting comments from github pr

Commit:
Merge branch 'nipy-dipy-master' into learn_dipy

matthieudumont committed:
Add the possibility to save a subset of metrics

jchoude committed:
Merge pull request #1120 from arokem/fix-1119

Fix 1119

RafaelNH committed:
TEST: Add multi-voxel test according to @arokem 's suggestion

arokem committed:
Merge pull request #1110 from sahmed95/ivim_linear

WIP : Ivim linear fitting

arokem committed:
Merge pull request #1124 from RafaelNH/fwdti_fix

Small fix in free water DTI model

kesshijordan committed:
addressing errors in test

kesshijordan committed:
Merge remote-tracking branch 'origin/master'

kesshijordan committed:
Merge remote-tracking branch 'upstream/master'

kesshijordan committed:
cleaning up

kesshijordan committed:
numpy.ufunc.at failed one build, so mod to conform

kesshijordan committed:
fixed type

matthieudumont committed:
Made a subclass for restore reconstruction

Commit:
Fix references and make PEP8 style corrections

matthieudumont committed:
Corrected typos, moved sigma to mandatory param

kesshijordan committed:
previous fix for np 1.7 introduced errors: addressed this

MrBago committed:
BF - fix _min_at function

matthieudumont committed:
ReconsT_dti tests

nilgoyette committed:
Cleaning and optimization

arokem committed:
Merge pull request #1060 from omarocegueda/fast_cc

Fast computation of Cross Correlation metric

kesshijordan committed:
Merge pull request #1 from MrBago/bf_minat

BF - fix _min_at function

arokem committed:
PEP8

arokem committed:
PEP8

arokem committed:
BF: Potential fix.

arokem committed:
BF: Uncomment globally set variables.

arokem committed:
Add this fix in newer version of scipy, just to be on the safe side.

Commit:
adding the Intensity adjustment function

Commit:
add Intensity adjustment and Upper_bound function

Commit:
pep8 formatting

Commit:
visualiSE -> ZE (US)

Garyfallidis committed:
Merge pull request #1034 from StongeEtienne/viz_surfaces

Viz surfaces

Commit:
add intensity adjustment test

Commit:
add two functions of intensity adjustment in threshold.py

Garyfallidis committed:
Merge pull request #1129 from arokem/fix-1127

Fix 1127

Commit:
test_visualization.py

Commit:
remove two files

Garyfallidis committed:
DOC: corrections in vtk.py

Garyfallidis committed:
DOC: corrections in viz/utils.py

Garyfallidis committed:
DOC: updated tutorial

Garyfallidis committed:
Linked viz_examples so it can be rendered in the website

Garyfallidis committed:
Example now renders correctly

Garyfallidis committed:
Merge pull request #1092 from nilgoyyou/target_line_based

Line-based target()

Garyfallidis committed:
Corrected typos

Commit:
modify the test_adjustment

MarcCote committed:
Merge pull request #1130 from Garyfallidis/viz_web

Minor corrections for showing surfaces

MarcCote committed:
Compressed recorded events files.

MarcCote committed:
Added coverage for VTK

Commit:
remove comment

MarcCote committed:
RF+TST

Garyfallidis committed:
Merge pull request #1125 from Tingyiwanyan/learn_dipy

Intensity adjustment. Find a better upper bound for interpolating images.

MarcCote committed:
BF in interactor tests

MarcCote committed:
BF: on mouse move events notify props only once.

MarcCote committed:
PEP + addressed @Garyfallidis's comments

MarcCote committed:
Remove coveralls badge. Only run build with COVERAGE=1 once. Drop build with python-vtk6.

MarcCote committed:
Trying to make mouse evebts work properly on Travis with Xvfb

MarcCote committed:
Still trying to make mouse events work properly on Travis with Xvfb

MarcCote committed:
Still trying to make mouse events work properly on Travis with Xvfb

MarcCote committed:
Changed recording

MarcCote committed:
Resort to not test MouseWheel events

MarcCote committed:
Added a class Event that contains information about the occuring VTK event.

MarcCote committed:
Resort to not test MouseWheel events

MarcCote committed:
RF: Refactored how the abort flag is set. Made it work on VTK 5.8 :-)

arokem committed:
PEP8

kesshijordan committed:
PEP8 compliance edits

kesshijordan committed:
Merge branch 'master' of git://github.com/nipy/dipy

Conflicts:
	dipy/tracking/tests/test_utils.py

kesshijordan committed:
Merge branch 'master' of github.com:kesshijordan/dipy

Conflicts:
	dipy/tracking/tests/test_utils.py

kesshijordan committed:
Merge branch 'master' of https://github.com/nipy/dipy

Conflicts:
	dipy/tracking/tests/test_utils.py

kesshijordan committed:
Merge branch 'master' of https://github.com/kesshijordan/dipy

Conflicts:
	dipy/tracking/tests/test_utils.py

arokem committed:
Merge pull request #1132 from MarcCote/change_travis_config

Update .travis.yml and README.md

arokem committed:
Merge pull request #1122 from matthieudumont/recons_dti_workflow

Add the reconst_dti workflow

arokem committed:
Merge pull request #1114 from kesshijordan/master

flexible grid to streamline affine generation and pathlength function

Commit:
Update deprecated function.  get_header() / get_affine() method is deprecated from nibabel 2.1, we use header / affine property instead

arokem committed:
Add a test speficically aimed at 0 evals ODFs.

arokem committed:
Now as a test.

Commit:
RF: Insure that fwdti nls procedure does not crash when nan are produced in eigenvalues decomposition

MarcCote committed:
Merge pull request #1064 from arokem/faster-dti-odf

Faster dti odf

MarcCote committed:
Mention new feature in docstring

MarcCote committed:
Merge pull request #1134 from arokem/fix-test

Rebase of https://github.com/nipy/dipy/pull/993

RafaelNH committed:
NF: add helper function to decompose tensors and dealing with nan tensors

RafaelNH committed:
RF: Non-linear DTI procedure uses the new helper function

RafaelNH committed:
TEST: test _decompose_tensor_nan

MarcCote committed:
Merge pull request #1113 from arokem/min-signal-context

TST: Test for invariance of model_params to splitting of the data.

RafaelNH committed:
RF: fwdti nls procedure uses new _decompose_tensor_nan function

arokem committed:
Merge pull request #1136 from skoudoro/example_update

Update deprecated function.

RafaelNH committed:
DOC: small corrections according to @arokem comments

RafaelNH committed:
BF: Fixed a bug in tests

Garyfallidis committed:
Merge pull request #1098 from MarcCote/enh_viz_renderer

Adding custom interactor for vizualisation

MarcCote committed:
Do not cover files related to benchmarks.

arokem committed:
Merge pull request #1141 from MarcCote/change_codecov_config

Do not cover files related to benchmarks.

gauvinalexandre committed:
Merge pull request #1 from MarcCote/viz_point_color_support

Mention new feature in docstring

arokem committed:
Merge pull request #942 from gauvinalexandre/viz_point_color_support

NF: Added support to colorize each line points indivdually

RafaelNH committed:
Changes DKI_signal to dki_signal in dipy.sims.voxel and respective tests

arokem committed:
Merge pull request #1137 from RafaelNH/fix_fwdti_eigenvalues

Small fix to insure that fwDTI non-linear procedure does not crash

Garyfallidis committed:
Merge pull request #1149 from RafaelNH/dki_signal

Change DKI_signal to dki_signal

nilgoyette committed:
Define fmin() for Visual Studio

arokem committed:
WIN32 => _WIN32

Commit:
Merge pull request #1 from arokem/broken_builds_fmin

WIN32 => _WIN32

arokem committed:
Merge pull request #1151 from nilgoyyou/broken_builds_fmin

Define fmin() for Visual Studio

Garyfallidis committed:
BF: closing matplotlib plots for each file

arokem committed:
Merge pull request #1158 from Garyfallidis/readd_plot_close

BF: closing matplotlib plots for each file while running the examples

kesshijordan committed:
Combined contour function with slicer to use affine

kesshijordan committed:
PEP8 compliance edits
